src/app/bav/controllers/root.js

```diff
@@ -0,0 +1,18 @@
+const { Controller: BaseController } = require("hmpo-form-wizard");
+
+class RootController extends BaseController {
+  async saveValues(req, res, next) {
+    const sharedClaims = req.session?.shared_claims;
+
+    if (sharedClaims) {
+
+      if (sharedClaims?.name?.length > 0) {
+        req.sessionModel.set("firstName", sharedClaims.name[0].nameParts[0].value);
+        req.sessionModel.set("surname", sharedClaims.name[0].nameParts[2].value);
```

Comment:
Will surname always be [2] even if the users has multiple middle names?

Reply:
Good catch - no, it is possible to have multiple given names, have changed the code to account for this - do we need to consider the same in CIC?

Reply:
CIC doesn't have names come through in shared_claims right?